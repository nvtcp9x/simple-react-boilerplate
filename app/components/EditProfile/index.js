import * as React from 'react';
import { StyledGrid } from '../../../utils/commonStyle'
import axios from 'axios'
import { connect } from 'react-redux'
import { saveUser } from '../../containers/Login/action'

type Props = {
  saveUser: () => void,
  picture: string,
  name: string,
  email: string,
  userInfo: Object,
}

class EditProfile extends React.Component<Props> {
  static defaultProps = {
    userInfo: {
      name: '',
      email: '',
      picture: '',
    },
  }

  state = {
    name: this.props.userInfo ? this.props.userInfo.name : '',
    email: this.props.userInfo ? this.props.userInfo.email : '',
    picture: this.props.userInfo ? this.props.userInfo.picture : '',
  }

  componentDidMount() {
    if (this.props.userInfo === EditProfile.defaultProps.userInfo) {
      axios.get('http://localhost:3000/users')
        .then(({ data }) => {
          const { name, email, picture } = data[0];
          const payload = {
            name,
            email,
            picture,
          }

          this.props.saveUser(payload)
          this.setState({
            name,
            email,
            picture,
          })
        })
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { name, email, picture } = this.state;
    const payload = {
      name,
      email,
      picture,
    }

    this.props.saveUser(payload)
    axios.get('http://localhost:3000/users').then(({ data }) => {
      axios.put(`http://localhost:3000/users/${data[0].id}`, {
        name,
        picture,
        email,
      })
    });
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { name, email, picture } = this.state;

    return (
      <StyledGrid>
        <h1>Edit Profile</h1>
        <form onSubmit={this.handleSubmit}>
          <label>Name:</label>
          <input type="text" name="name" onChange={this.handleChange} value={name} />
          <label>Email:</label>
          <input type="email" name="email" onChange={this.handleChange} value={email} />
          <input type="submit" value="Submit" />
        </form>
      </StyledGrid>
    )
  }
}

const mapStateToProps = state => ({
  userInfo: state.userReducer.payload,
})

export default connect(mapStateToProps, { saveUser })(EditProfile)
