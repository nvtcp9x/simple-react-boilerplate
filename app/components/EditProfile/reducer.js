import { UPDATE_USER, UPDATE_USER_SUCCESS, UPDATE_USER_FAIL } from './constant'

const initialState = {
  isUpdating: false,
  user: {},
  error: {},
}

export function editUserReducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_USER:
      return {
        ...state,
        isUpdating: true,
      }
    case UPDATE_USER_SUCCESS:
      return {
        ...state,
        isUpdating: false,
        user: action.data,
      }
    case UPDATE_USER_FAIL:
      return {
        ...state,
        isUpdating: false,
        error: action.error,
      }
    default:
      return state;
  }
}
