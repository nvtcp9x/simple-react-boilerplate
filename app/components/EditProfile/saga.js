import { takeLatest, put, call } from 'redux-saga/effects';
import { UPDATE_USER } from './constant';
import { updateUserApi } from './service'
import { updateUserSucess, updateUserFail } from './action';

export function* updateUser(action) {
  try {
    const { data } = action;
    const response = yield call(updateUserApi, data);
    yield put(updateUserSucess(response.data));
  } catch (err) {
    yield put(updateUserFail(err));
  }
}

export function* watchUpdateUser() {
  yield takeLatest(UPDATE_USER, updateUser)
}
