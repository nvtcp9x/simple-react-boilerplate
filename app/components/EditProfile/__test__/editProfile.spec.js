import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import EditProfile from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('EditProfile component', () => {
  const component = shallow(<EditProfile />);

  it('should match to SnapShot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
