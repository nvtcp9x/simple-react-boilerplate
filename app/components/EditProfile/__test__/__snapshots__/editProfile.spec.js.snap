// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile component should match to SnapShot 1`] = `
<React.Fragment>
  <withRouter(Connect(Header)) />
  <Styled(Container)>
    <Connect(UserForm) />
  </Styled(Container)>
  <Footer />
</React.Fragment>
`;
